test/modules/tiledimage.js

```diff
@@ -132,6 +135,8 @@
     QUnit.test('update', function(assert) {
         var done = assert.async();
         var handlerCount = 0;
+        var testTileDrawingEvent = viewer.drawer.getType() === 'canvas';
```

Comment:
It seems odd that we aren't controlling what kind of drawer we run here, but instead are just checking. Should we run this test once for each kind of drawer?

Reply:
I'm thinking we should run ALL of the tests for all of the different drawers, and only skip them when it doesn't make sense for that drawer type.

In this case, only the `canvas` drawer fires the event, so if a different drawer is being used it doesn't make sense to listen for that event.

Reply:
Running all the tests for all the drawers sounds good... I expect going forward we will be more likely to end up with something that works in one drawer but not another. That said, surely some tests are unrelated to drawers entirely... I guess the question is whether it's worth singling those things out or whether it's better policy to just do it for everyone (knowing that makes our tests run 3x as long).

Reply:
I ended up just running the tests in the `drawer` module multiple times, when it makes sense for each viewer. The test here actually is of the logic of the `TiledImage` class, which doesn't depend on which specific drawer is used. So, I made this test always use the `canvas` drawer so that the original test works like it always did.